qiskit/pulse/utils.py

```diff
@@ -0,0 +1,113 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Pulse utilities.
+"""
+import warnings
+
+from typing import List
+
+from .channels import AcquireChannel, MemorySlot
+from .commands import Acquire, AcquireInstruction
+from .interfaces import ScheduleComponent
+from .schedule import Schedule
+from .cmd_def import CmdDef
+
+
+def align_measures(schedule: ScheduleComponent, cmd_def: CmdDef) -> Schedule:
+    """Return a new schedule where measurements occur at the same physical time, with the remaining
+    schedules appropriately offset. Minimum measurement wait time (to allow for calibration pulses)
+    is enforced.
+    This is only defined for schedules that are acquire-less or acquire-final per channel: a
+    schedule with pulses or acquires occuring on a channel which has already had an acquire will
+    throw an error.
+
+    Args:
+        schedule: Schedule to be aligned
+        cmd_def: Command definition list
+    Returns:
+        Schedule
+    Raises:
+        ValueError: if an acquire or pulse is encountered on a channel that has already been part
+                    of an acquire
+    """
+    new_schedule = Schedule()
+
+    # Need time to allow for calibration pulses to be played for result classification
+    max_calibration_duration = 0
+    for qubits in cmd_def.cmd_qubits('u2'):
+        cmd = cmd_def.get('u2', qubits)
```

Comment:
`u2` will typically have a single parameter that will need to be filled to get the schedule back. `CmdDef.get_parameters` can be used to get the required parameters and then you can fill these parameters with 0s to extract a schedule for `u2` which will allow you to get the required duration.

Reply:
Rather this should also be `u3(np.pi, 0, np.pi)` as this is an `x` gate.

Reply:
I made the change but I think something I did in the `test/mock.py` is wrong or missing since changing the parameters doesn't seem to have any effect.

Reply:
I agree. I think its because the examples in mock are not proper u1, u2, u3, gate etc. For example `u3` should have `P1`, `P2`, `P3` present. I am surprised this is not failing when you are providing extra arguments? If so this may be a bug in the parameter handling.

Reply:
Yeah, it always seems to run fine no matter what parameters are provided.

Reply:
This is now being tracked by  #2603 and will be fixed by #2493.